### DISCUSSION TITLE
Access filtered rows from other components

### DISCUSSION THREAD

**Discussion by qwasyx0** [rappasoft/laravel-livewire-tables#331] [Q&A]
Hello, I'm still learning with Livewire (first project) and sorry for my bad english. I am looking for a way of getting filtered rows or whole Builder query to another component (graphs) so they can redraw based on the applied filters by user. I am able to get the query displayed in dd() inside the components but can't seem to find a way of accessing it externally.

> public function query(): Builder
    {
        $query = IRSVOZIKY::query()
            ->when($this->getFilter('ZNACKA'), fn ($query, $filter) => $query->where('ZNACKA', $filter))
            ->when($this->getFilter('ROKVYROBY'), fn ($query, $filter) => $query->where('ROKVYROBY', $filter))
            ->when($this->getFilter('UMISTENI'), fn ($query, $filter) => $query->where('UMISTENI', $filter))
            ->where('IDCISELPOD', Auth::user()->IDCISELPOD);
             //dd($query);                              <- this shows the whole query with wheres
            $this->emit('nakladyQueryChanged', $query);            <- listening to this event in other component returns empty query
            return $query;
    }

I also found out function getRowsProperty() which returns all the rows, which would be enough, but I don't know how to access these from the other graph component. Is there a way to hook on a refresh event with all the data?

**Comment by rappasoft** [comment]
In your query function you can probably emit an event with the current query since any time something is filtered it will get called.

**Reply by qwasyx0** [reply]
That's exactly what I tried in the code, emit right before returning the query, but the query i get where I'm listening is empty.

EDIT: I was able to get the query with $this->emit('nakladyQueryChanged', $query->getQuery()); which returns Builder instance with all needed data and $this->emit('nakladyQueryChanged', $query->get()); which returns unpaginated, unsorted and unfiltered rows as array.
